src/looper.js

```diff
@@ -1,520 +1,89 @@
-      var t = time || 0;
-      this.metro.start(t);
-    }
-  };
-
-  /**
-   *  Loop playback of this part. It will begin
-   *  looping through all of its phrases at a speed
-   *  determined by setBPM.
-   *
-   *  @method  loop
-   *  @param  {Number} [time] seconds from now
-   */
-  p5.Part.prototype.loop = function(time) {
-    this.looping = true;
-    // rest onended function
-    this.onended = function() {
-      this.partStep = 0;
-      // dont start phrases over, right?
-    };
-    var t = time || 0;
-    this.start(t);
-  };
-
-  /**
-   *  Tell the part to stop looping.
-   *
-   *  @method  noLoop
-   */
-  p5.Part.prototype.noLoop = function( ) {
-    this.looping = false;
-    // rest onended function
-    this.onended = function() {
-      this.stop();
-    };
-  };
-
-  /**
-   *  Stop the part and cue it to step 0.
-   *
-   *  @method  stop
-   *  @param  {Number} [time] seconds from now
-   */
-  p5.Part.prototype.stop = function(time) {
-    this.partStep = 0;
-    this.pause(time);
-  };
 
-  /**
-   *  Pause the part. Playback will resume
-   *  from the current step.
-   *
-   *  @method  pause
-   *  @param  {Number} time seconds from now
-   */
-  p5.Part.prototype.pause = function(time) {
-    this.isPlaying = false;
-    var t = time || 0;
-    this.metro.stop(t);
-  };
-
-  /**
-   *  Add a p5.Phrase to this Part.
-   *
-   *  @method  addPhrase
-   *  @param {p5.Phrase}   phrase   reference to a p5.Phrase
-   */
-  p5.Part.prototype.addPhrase = function(name, callback, array) {
-    var p;
-    if (arguments.length === 3) {
-      p = new p5.Phrase(name, callback, array);
-    } else if (arguments[0] instanceof p5.Phrase) {
-      p = arguments[0];
-    } else {
-      throw 'invalid input. addPhrase accepts name, callback, array or a p5.Phrase';
-    }
-    this.phrases.push(p);
-
-    // reset the length if phrase is longer than part's existing length
-    if (p.sequence.length > this.length) {
-      this.length = p.sequence.length;
-    }
-  };
+    this.callback = callback;
+    this._interval = interval;
+    this._timeSignature = 4;
 
-  /**
-   *  Remove a phrase from this part, based on the name it was
-   *  given when it was created.
-   *
-   *  @method  removePhrase
-   *  @param  {String} phraseName
-   */
-  p5.Part.prototype.removePhrase = function(name) {
-    for (var i in this.phrases) {
-      if (this.phrases[i].name === name) {
-        this.phrases.splice(i, 1);
-      }
-    }
-  };
+    this.bpm = BPM || 60;
 
-  /**
-   *  Get a phrase from this part, based on the name it was
-   *  given when it was created. Now you can modify its array.
-   *
-   *  @method  getPhrase
-   *  @param  {String} phraseName
-   */
-  p5.Part.prototype.getPhrase = function(name) {
-    for (var i in this.phrases) {
-      if (this.phrases[i].name === name) {
-        return this.phrases[i];
-      }
-    }
+    this.interations;
+    this.clock = new Clock({
+      "callback" : this.callback,
```

Comment:
I think we may want/need to update the callback to bind its scope (`this` within the context of the callback) to the p5.Looper, rather than to the Click. We can do this by creating a reference to this instance of the looper class (the variable name `self` is a convention but it could be anything), and then calling `self.callback`

We may also need to do this in order to pass in "time from now" to the callback, because that is what all existing p5.sound functions that accept a time are expecting. Tone follows the Web Audio convention of passing in time from the beginning of the web audio context.

```
    var self = this;
    this.clock = new Clock({
      "callback" : function(time) {
        var timeFromNow = time - p5sound.audiocontext.currentTime;
        self.callback(timeFromNow);
      },
```